### PR TITLE
Login Page Skeleton

### PR DESCRIPTION
**Pull request by achandrabalan**
## Notion task link

[Login Page Skeleton](https://www.notion.so/uwblueprintexecs/Login-Page-Skeleton-d58ed370f9374ee088ea688264b59bf0?pvs=4)

## Implementation description
* Rewrote all components in Chakra UI for easier design + app wide consistency
* Matched the Figma designs found [here](https://www.figma.com/file/woRXyIloJYllaqE62wjgYv/Login%2F2FA?type=design&node-id=608-2&mode=design&t=eD8rGdu8rrzRXzSy-0)

## Steps to test
1. Launch app, if you are brought to homepage click 'Logout'. 
2. You should be able to login with a correct user / pass combination
3. You should not be able to login with an incorrect user / pass combination
4. You should be able to visit the Sign Up page

## What should reviewers focus on?
* I would like to know how this design scales on different size monitors / displays. Currently all containers are dynamically sized with user's viewport. However, all text sizes have been hardcoded with pixel values.

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [ ] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [ ] I have run the appropriate linter(s)
- [ ] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR
- [ ] If I have made API changes, I have updated the [REST API Docs](https://www.notion.so/uwblueprintexecs/REST-Endpoints-05ce60312bb943439dfda42bb1318536)
- [ ] IF I have made changes to the db/models, I have updated the [Data Models Page](https://www.notion.so/uwblueprintexecs/Data-Models-760f8aa06b244eb0842c079ad77987b0)
- [ ] I have documented this PR in the [Production Release Page](https://www.notion.so/uwblueprintexecs/fe28a97bfa5648d3bc3795b32b694acd?v=5565cfb35dcc45bb84f6528cb09517cd)
- [x] I have updated other Docs as needed

### CODE REVIEW

File: frontend/src/components/auth/Credentials.tsx
Comment (phamkelly17) [inline frontend/src/components/auth/Credentials.tsx:155]: if you look at `supportive-housing/frontend/src/theme/common/buttonStyles.tsx` you'll see our design system and the button stylings we have on there. I think you should be able to design this styling as a new variant and then you can use that styling here

File: frontend/src/components/auth/Credentials.tsx
Comment (phamkelly17) [inline frontend/src/components/auth/Credentials.tsx:128]: see comment about the button styling and same thing for input

File: frontend/src/components/auth/Credentials.tsx
Comment (phamkelly17) [inline frontend/src/components/auth/Credentials.tsx]: same for text

**Review by phamkelly17 — CHANGES_REQUESTED** [review]
i reset my db yesterday so now i'm having issues signing out so i actually wasn't able to look at the new login page. but definitely should get the components used here in our design system

File: frontend/src/components/auth/Credentials.tsx
Comment (owen-sellner) [inline frontend/src/components/auth/Credentials.tsx]: The text colour and the placeholder colour should be different. I am pretty sure the text should be "#535353" and the placeholder is "#989898" like you have already.

File: frontend/src/components/auth/Credentials.tsx
Comment (owen-sellner) [reply frontend/src/components/auth/Credentials.tsx:155]: ^^^ for the button styling you will want to use the values from `supportive-housing/frontend/src/theme/colors.tsx`. If you want to see the colours there is a figma page for it [here](https://www.figma.com/file/6Q5yT8sRQbv6VUR8hoj8ET/SHOW-Chakra-UI?type=design&node-id=1-2&mode=design&t=cVYqY9DXh6pmy77a-0)

File: frontend/src/components/auth/Credentials.tsx
Comment (owen-sellner) [inline frontend/src/components/auth/Credentials.tsx:127]: I think it will be easier to set up the error checking later on if this was an event handler method instead.

**Review by owen-sellner — DISMISSED** [review]
You matched the design really well!! I left a few comments about some things that should be addressed before merging.

File: frontend/src/components/auth/Credentials.tsx
Comment (kevin-pierce) [reply frontend/src/components/auth/Credentials.tsx:127]: This PR aims to mainly address the styling ; I'm addressing the error message here: https://github.com/uwblueprint/supportive-housing/pull/175/files

**Review by phamkelly17 — APPROVED** [review]
LGTM ^_^ one thing that annoys me in when you hover over the "Login" button, the entire button turns white which just blends in with the page. i'm not gonna block your PR over that though but i vote we just change the button colour to the dark green or something

File: frontend/src/components/auth/Credentials.tsx
Comment (connor-bechthold) [inline frontend/src/components/auth/Credentials.tsx]: nit: This newline can be removed

**Review by connor-bechthold — DISMISSED** [review]
This looks BEAUTIFUL I'm so glad we're finally retiring the default starter code design 10 months later 😎 

Going off of what Kelly was saying above, you can do something similar to the add log record button for the hover effect (I'm not sure if this is an issue on my end but the login button didn't do anything when I hovered over it and my cursor never changed to a pointer 🤔 ):

https://github.com/uwblueprint/supportive-housing/assets/64047863/a08e34e0-b2ee-40ae-8be1-f32f44d6dafb

Also, I'd like if the `Sign Up Now` link had some indication that it was a link. Right now when you hover over it (not sure if this an issue on just my end), it's just normal text. Ideally, when hovering over it, the cursor should change to a pointer and the the text itself should be underlined (indicating it's a link).

Let me know if you have any questions/concerns with any of that!

**Comment by achandrabalan** [comment]
> LGTM ^_^ one thing that annoys me in when you hover over the "Login" button, the entire button turns white which just blends in with the page. i'm not gonna block your PR over that though but i vote we just change the button colour to the dark green or something

hmm I ran into this issue before too. I made a change to fix this, do you mind checking it out again?

**Comment by achandrabalan** [comment]
> This looks BEAUTIFUL I'm so glad we're finally retiring the default starter code design 10 months later 😎
> 
> Going off of what Kelly was saying above, you can do something similar to the add log record button for the hover effect (I'm not sure if this is an issue on my end but the login button didn't do anything when I hovered over it and my cursor never changed to a pointer 🤔 ):
> 
>  Screen.Recording.2023-10-09.at.2.32.17.AM.mov 
> Also, I'd like if the `Sign Up Now` link had some indication that it was a link. Right now when you hover over it (not sure if this an issue on just my end), it's just normal text. Ideally, when hovering over it, the cursor should change to a pointer and the the text itself should be underlined (indicating it's a link).
> 
> Let me know if you have any questions/concerns with any of that!

I think the major problem was because the button needs to be disabled, I looked into it and chakra UI has some default effects for disabled buttons that you have to manually override. I made the changes you requested let me know if it got lost in translation.

**Comment by phamkelly17** [comment]
> > LGTM ^_^ one thing that annoys me in when you hover over the "Login" button, the entire button turns white which just blends in with the page. i'm not gonna block your PR over that though but i vote we just change the button colour to the dark green or something
> 
> hmm I ran into this issue before too. I made a change to fix this, do you mind checking it out again?

login button lgtm

File: frontend/src/components/auth/Credentials.tsx
Comment (connor-bechthold) [inline frontend/src/components/auth/Credentials.tsx:145]: So this is a super small nitpick but when the button is disabled and you hover over it, the cursor turns into a pointer, which kind of gives the illusion that it's clickable (even though it's not). You can slightly modify this to change that behaviour:
```suggestion
                _hover={
                  email && password
                    ? {
                        background: "teal.500",
                        transition:
                          "transition: background-color 0.5s ease !important",
                      }
                    : {}
                }
```

Essentially when we have both the username and password, the hover effect should occur, otherwise, there shouldn't be an effect at all.

**Review by connor-bechthold — APPROVED** [review]
Works perfectly! Had one nitpick to address but this should be good to merge after. GREAT WORK 👨‍🍳 👨‍🍳 👨‍🍳

**Pull request merged**
